source/Handlebars/MemberAccessors/ReflectionMemberAccessor.cs

```diff
@@ -92,20 +92,37 @@ public RawObjectTypeDescriptor(Type type)
                         .Invoke(null, new object[] {property});
                 }
 
-                var field = type.GetFields(BindingFlags.Instance | BindingFlags.Public).FirstOrDefault(o => string.Equals(o.Name, name.LowerInvariant, StringComparison.OrdinalIgnoreCase));
-                
+                var field = type.GetFields(BindingFlags.Instance | BindingFlags.Public).FirstOrDefault(o => string.Equals(o.Name, name.LowerInvariant, StringComparison.OrdinalIgnoreCase));
+
                 if (field != null)
                 {
                     return o => field.GetValue(o);
                 }
 
                 return null;
-            }
+            }
+
+            delegate TValue ValueTypeGetterDelegate<T, TValue>(ref T instance);
```

Comment:
nit: Consider moving delegate type declaration to the beginning of the class and declare it explicitly `private`.